corehq/apps/registration/utils.py

```diff
@@ -144,9 +146,31 @@ def request_new_domain(request, project_name, is_new_user=True, is_new_sso_user=
     dom_req.domain = new_domain.name
 
     if not settings.ENTERPRISE_MODE:
-        _setup_subscription(new_domain.name, current_user)
+        try:
+            _setup_subscription(new_domain.name, current_user)
+        except Exception as error:
+            notify_exception(request, "Error initializing subscription for new domain", details={
+                'domain': new_domain.name,
+                'hr_name': project_name,
+                'creating_user': current_user.username,
+                'first_domain_for_user': is_new_user,
+                'error': str(error),
+            })
+            new_domain.delete()
```

Comment:
@dannyroberts do you think there is an issue deleting the domain here [if this bit of code](https://github.com/dimagi/commcare-hq/blob/bf4655c601521de99b06f5730c3b5acd5ec40ffd/corehq/apps/accounting/models.py#L1187) that's run in the `save` method for `Subscription` throws a `NoBrokerAvailable` error if kafka is down?